### PR TITLE
[GCS] Change GCS Test to cc_test

### PR DESCRIPTION
**Pull request by micafan**
## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -811,43 +811,63 @@ cc_library(
 )
 
 # TODO(micafan) Replace cc_binary with cc_test for GCS test.
```
Comment (raulchen) [inline BUILD.bazel]: remove this TODO?

File: src/ray/util/test_util.h [hunk 2]
```diff
@@ -58,6 +59,34 @@ std::shared_ptr<RayObject> GenerateRandomObject() {
...
+/// Path to redis server executable binary.
+std::string REDIS_SERVER_EXEC_PATH;
+/// Path to redis client executable binary.
+std::string REDIS_CLIENT_EXEC_PATH;
+/// Path to redis module library.
+std::string REDIS_MODULE_LIBRARY_PATH;
+
+/// Test helper class, it will start redis server before the test runs,
+/// and stop redis server after the test is completed.
+class ManageRedisServiceForTest : public ::testing::Test {
```
Comment (raulchen) [inline src/ray/util/test_util.h]: can you also change `GcsServerTest` to use this base class?

**Review by raulchen — APPROVED** [review]

File: BUILD.bazel (hunk 1 above)
Comment (micafan) [reply BUILD.bazel]: I will change this to do, not just remove it. bazel seems support test group, which can run UT as a batch. I think it is useful for developer. Will support this sooner.

File: src/ray/util/test_util.h (hunk 2 above)
Comment (micafan) [reply src/ray/util/test_util.h]: Of course, that's part of the reason why this PR is still WIP.

File: BUILD.bazel (hunk 1 above)
Comment (raulchen) [reply BUILD.bazel]: sounds good

File: src/ray/util/test_util.h [hunk 3]
```diff
@@ -58,6 +60,34 @@ std::shared_ptr<RayObject> GenerateRandomObject() {
   return std::shared_ptr<RayObject>(new RayObject(GenerateRandomBuffer(), nullptr));
 }
 
+/// Path to redis server executable binary.
+std::string REDIS_SERVER_EXEC_PATH;
+/// Path to redis client executable binary.
+std::string REDIS_CLIENT_EXEC_PATH;
+/// Path to redis module library.
+std::string REDIS_MODULE_LIBRARY_PATH;
```
Comment (zhijunfu) [inline src/ray/util/test_util.h]: Can we put these above variables as public fields of `ManageRedisServiceForTest` instead?

Right now they are referenced as `ray::REDIS_SERVER_EXEC_PATH,` it's not clear that they are actually only for tests.

File: src/ray/util/test_util.h [hunk 4]
```diff
@@ -58,6 +60,34 @@ std::shared_ptr<RayObject> GenerateRandomObject() {
...
+
+/// Test helper class, it will start redis server before the test runs,
+/// and stop redis server after the test is completed.
+class ManageRedisServiceForTest : public ::testing::Test {
```
Comment (zhijunfu) [inline src/ray/util/test_util.h]: How about `RedisServerTestWrapper` instead of `ManageRedisServiceForTest` ?

**Review by zhijunfu — APPROVED** [review]
thanks. I added a few comments.

File: src/ray/util/test_util.h (hunk 3 above)
Comment (micafan) [reply src/ray/util/test_util.h]: Sounds good.

File: src/ray/util/test_util.h (hunk 4 above)
Comment (micafan) [reply src/ray/util/test_util.h]: `RedisServerTestWrapper` sounds like the test for `RedisServer`, right?

File: src/ray/util/test_util.h (hunk 3 above)
Comment (micafan) [reply src/ray/util/test_util.h]: I just recall why i didn't make those `vars` as `member` of `ManageRedisServiceForTest`. Those `vars` will used in `gtest` static function `SetUpTestCase` to setup redis service. So those `vars` only can be `static variable` or `global variable`.  Obviously `static variable` not ok.

**Pull request merged**